src/swagger/schemas/common.yml

```diff
@@ -7,91 +7,238 @@ api-ref-model:
       type: string
     name:
-      description: "Name of the referenced resource."
+      description: 'Name of the referenced resource.'
       type: string
     url:
-      description: "URL of the referenced resource."
+      description: 'URL of the referenced resource.'
       type: string
 api-ref-list-model:
   description: |
     `APIReferenceList`
   type: object
   properties:
     count:
-      description: "Total number of resources available."
+      description: 'Total number of resources available.'
       type: number
     results:
       type: array
       items:
-        $ref: "./combined.yml#/APIReference"
-choice-model:
+        $ref: './combined.yml#/APIReference'
+damage-model:
   description: |
-    `Choice`
+    `Damage`
   type: object
   properties:
-    choose:
-      description: "Number of items to pick from the list."
+    damage_dice:
+      type: string
+    damage_type:
+      $ref: './combined.yml#/APIReference'
+dc-model:
+  description: |
+    `DC`
+  type: object
+  properties:
+    dc_type:
+      $ref: './combined.yml#/APIReference'
+    dc_value:
+      description: 'Value to beat'
       type: number
-    type:
-      description: "Type of the resources to choose from."
+    success_type:
+      description: 'Result of a successful save. Can be \"none\", \"half\", or \"other\"'
       type: string
-    from:
-      description: "List of resources to choose from."
-      type: array
-      items:
-        $ref: "./combined.yml#/APIReference"
-string-choice-model:
+option-model:
+  description: |
+    `Option`
+  oneOf:
+    - type: object
+      properties:
+        option_type:
+          description: 'Type of option; determines other attributes.'
+          type: string
+        item:
+          $ref: './combined.yml#/APIReference'
+    - type: object
+      properties:
+        option_type:
+          description: 'Type of option; determines other attributes.'
+          type: string
+        action_name:
+          description: 'The name of the action.'
+          type: string
+        count:
+          description: 'The number of times this action can be repeated if chosen.'
+          type: number
+        type:
+          description: 'For attack options that can be melee, ranged, abilities, or thrown.'
+          type: string
+          enum: [melee, ranged, ability, magic]
+    - type: object
+      properties:
+        option_type:
+          description: 'Type of option; determines other attributes.'
+          type: string
+        items:
+          type: array
+          items:
+            $ref: './combined.yml#/APIReference' # Should be Option but can't use due to circular reference
+    - type: object
+      properties:
+        option_type:
+          description: 'Type of option; determines other attributes.'
+          type: string
+        choice:
+          $ref: './combined.yml#/APIReference' # Should be Choice but can't use due to circular reference
+    - type: object
```

Comment:
@bagelbits @fergcb I'd like your input on how to resolve this. In order to accurately represent the structure of the data from the database change, I would need to use a circular reference. However, swagger/openapi doesn't support circular references. How do you think we should resolve this contradiction?

Reply:
@bagelbits @fergcb I managed to resolve the issue by changing the import of `swagger-cli` that gets used. Instead of using the package `swagger-cli`, the project now uses `@apidevtools/swagger-cli`.